core: Triggering error when receipts count does not match with transactions count in the block

wraith7 opened the pull request:
Triggering error when receipts count does not match with transactions count in the block in SetReceiptsData function.

karalabe commented on core/blockchain.go:
```diff
@@ -723,10 +723,16 @@ func (bc *BlockChain) Rollback(chain []common.Hash) {
+			len(transactions), "len(receipts)", len(receipts))
+		return fmt.Errorf("length of transactions in block does not match with the length of receipts: len(transaction) is %d, len(receipts) is %d",
+			len(transactions), len(receipts))
+	}
```
Please just `return errors.New("transaction and receipt count mismatch")` , also no need for an explicit error log.

wraith7 replied:
Done

karalabe commented on core/blockchain.go:
```diff
@@ -742,7 +748,7 @@ func SetReceiptsData(config *params.ChainConfig, block *types.Block, receipts ty
 			receipts[j].GasUsed = receipts[j].CumulativeGasUsed
 		} else {
-			receipts[j].GasUsed = receipts[j].CumulativeGasUsed - receipts[j-1].CumulativeGasUsed
+			receipts[j].GasUsed = receipts[j].CumulativeGasUsed - receipts[j - 1].CumulativeGasUsed
```
Please revert this (gofmt).

wraith7 replied:
Done

wraith7 commented:
Addressed code review.

One quick question.
When I got 'requested changes', should I revert commit and address code review by overwriting commit or add another commit on top of existing commit?

holiman commented:
You can either squash/rebase (as you did), or make another commit. Peter likes to squash-merge, so it'll be only one commit on master either way

wraith7 commented:
Thanks a lot for the feedback.

karalabe commented:
If there's a single commit that is signed, I merge. If there are multiple commits or one without a signature, I squash :)
